src/Components/Aspire.Seq/AspireSeqExtensions.cs

```diff
@@ -23,47 +24,41 @@ public static class AspireSeqExtensions
+        Action<SeqSettings>? configureSettings = null)
     {
         ArgumentNullException.ThrowIfNull(builder);
 
         var settings = new SeqSettings();
+        settings.ServerUrl = builder.Configuration.GetConnectionString(connectionName);
+        settings.Logs.Protocol = OtlpExportProtocol.HttpProtobuf;
+        settings.Traces.Protocol = OtlpExportProtocol.HttpProtobuf;
+
         builder.Configuration.GetSection("Aspire:Seq").Bind(settings);
+        configureSettings?.Invoke(settings);
 
-        if (builder.Configuration.GetConnectionString(connectionName) is string connectionString)
+        if (!string.IsNullOrEmpty(settings.ServerUrl))
         {
-            settings.ServerUrl = connectionString;
+            settings.Logs.Endpoint = new Uri($"{settings.ServerUrl}/ingest/otlp/v1/logs");
+            settings.Traces.Endpoint = new Uri($"{settings.ServerUrl}/ingest/otlp/v1/traces");
         }
-
-        configureSettings?.Invoke(settings);
-
-        if (string.IsNullOrEmpty(settings.ServerUrl))
+        if (!string.IsNullOrEmpty(settings.ApiKey))
         {
-            settings.ServerUrl = "http://localhost:5341";
+            settings.Logs.Headers = string.IsNullOrEmpty(settings.Logs.Headers) ? $"X-Seq-ApiKey={settings.ApiKey}" : $"{settings.Logs.Headers},X-Seq-ApiKey={settings.ApiKey}";
+            settings.Traces.Headers = string.IsNullOrEmpty(settings.Traces.Headers) ? $"X-Seq-ApiKey={settings.ApiKey}" : $"{settings.Traces.Headers},X-Seq-ApiKey={settings.ApiKey}";
         }
 
-        builder.Services.Configure<OpenTelemetryLoggerOptions>(logging => logging.AddOtlpExporter(opt =>
-        {
-            opt.Endpoint = new Uri($"{settings.ServerUrl}/ingest/otlp/v1/logs");
-            opt.Protocol = OtlpExportProtocol.HttpProtobuf;
-            if (!string.IsNullOrEmpty(settings.ApiKey))
-            {
-                opt.Headers = $"X-Seq-ApiKey={settings.ApiKey}";
-            }
-        }));
-        builder.Services.ConfigureOpenTelemetryTracerProvider(tracing => tracing
-            .AddOtlpExporter(opt =>
-                {
-                    opt.Endpoint = new Uri($"{settings.ServerUrl}/ingest/otlp/v1/traces");
-                    opt.Protocol = OtlpExportProtocol.HttpProtobuf;
-                    if (!string.IsNullOrEmpty(settings.ApiKey))
-                    {
-                        opt.Headers = $"X-Seq-ApiKey={settings.ApiKey}";
-                    }
-                }
+        builder.Services.Configure<OpenTelemetryLoggerOptions>(logging => logging.AddProcessor(
+            sp => new BatchLogRecordExportProcessor(new OtlpLogExporter(settings.Logs))
+            ));
+
+        builder.Services.ConfigureOpenTelemetryTracerProvider(tracing => tracing.AddProcessor(
+            sp => new BatchActivityExportProcessor(new OtlpTraceExporter(settings.Traces))
             ));
 
-        if (settings.HealthChecks)
+        if (settings is { HealthChecks: true, ServerUrl: not null })
```

Comment:
Should it be an error if you haven't set ServerUrl? Seems odd that setting `HealthChecks = true`, but not getting health checks would be silent.

Reply:
I don't think it should be an error because a user could set `Logs.EndPoint` or `Traces.EndPoint` directly. 

The combination of no `ServerUrl` and `HealthChecks == true` is never valid, so I will make that an error.